core/include/scipp/core/transform.h

```diff
@@ -25,207 +25,14 @@
-constexpr auto operator-(const ValueAndVariance<T1> a,
-                         const ValueAndVariance<T2> b) noexcept {
-  return ValueAndVariance{a.value - b.value, a.variance - b.variance};
-}
-template <class T1, class T2>
-constexpr auto operator*(const ValueAndVariance<T1> a,
-                         const ValueAndVariance<T2> b) noexcept {
-  return ValueAndVariance{a.value * b.value,
-                          a.variance * b.value * b.value +
-                              b.variance * a.value * a.value};
-}
-template <class T1, class T2>
-constexpr auto operator/(const ValueAndVariance<T1> a,
-                         const ValueAndVariance<T2> b) noexcept {
-  return ValueAndVariance{
-      a.value / b.value,
-      (a.variance + b.variance * (a.value * a.value) / (b.value * b.value)) /
-          (b.value * b.value)};
-}
-
-template <class T1, class T2>
-constexpr auto operator+(const ValueAndVariance<T1> a, const T2 b) noexcept {
-  return ValueAndVariance{a.value + b, a.variance};
-}
-template <class T1, class T2>
-constexpr auto operator+(const T2 a, const ValueAndVariance<T1> b) noexcept {
-  return ValueAndVariance{a + b.value, b.variance};
-}
-template <class T1, class T2>
-constexpr auto operator-(const ValueAndVariance<T1> a, const T2 b) noexcept {
-  return ValueAndVariance{a.value - b, a.variance};
-}
-template <class T1, class T2>
-constexpr auto operator-(const T2 a, const ValueAndVariance<T1> b) noexcept {
-  return ValueAndVariance{a - b.value, b.variance};
-}
-template <class T1, class T2>
-constexpr auto operator*(const ValueAndVariance<T1> a, const T2 b) noexcept {
-  return ValueAndVariance{a.value * b, a.variance * b * b};
-}
-template <class T1, class T2>
-constexpr auto operator*(const T1 a, const ValueAndVariance<T2> b) noexcept {
-  return ValueAndVariance{a * b.value, a * a * b.variance};
-}
-template <class T1, class T2>
-constexpr auto operator/(const ValueAndVariance<T1> a, const T2 b) noexcept {
-  return ValueAndVariance{a.value / b, a.variance / (b * b)};
-}
-template <class T1, class T2>
-constexpr auto operator/(const T1 a, const ValueAndVariance<T2> b) noexcept {
-  return ValueAndVariance{a / b.value, b.variance * a * a /
-                                           (b.value * b.value) /
-                                           (b.value * b.value)};
-}
-
-/// Deduction guide for class ValueAndVariances. Using decltype to deal with
-/// potential mixed-type val and var arguments arising in binary operations
-/// between, e.g., double and float.
-template <class T1, class T2>
-ValueAndVariance(const T1 &val, const T2 &var)
-    ->ValueAndVariance<decltype(T1() + T2())>;
-
-/// A values/variances pair based on references to sparse data containers.
-///
-/// This is a helper for implementing operations for sparse container such as
-/// `clear`, and for descending into the sparse container itself, using a nested
-/// call to an iteration function.
-template <class T> struct ValuesAndVariances {
```

Comment:
I think this should stay in this file, it is an implementation detail of `transform`.